### ISSUE TITLE
Research setting up automated testing with Selenium 

### ISSUE THREAD

**Issue by alawvt** [VTUL/vtechworks#434]
See https://github.com/VTUL/vtechworks/wiki/06.-Testing for a list of old Selenium tests.

**Comment by alawvt** [comment]
Please try both programming and keyboard recording methods and report your results (time required per test, difficulty, etc.) in wiki page linked above.

**Comment by pgray64** [comment]
I checked out Keith's vtechworks_selenium_tests repo that uses the Firefox plugin with GUI. Tests are easy to set up and modify and version control is fine. 

I also set up testing with Nightwatch.js, which is a code based selenium testing framework. The benefit of this is that it can be configured to work with any browser. It also offers finer control in creating tests. I wrote a setup script so everything is set up once you run install.sh. (I only configured it for Firefox so far).
https://git.it.vt.edu/philippe/nightwatch-test

Both the Firefox plugin and Nightwatch.js take about the same amount of time to write unit tests and run them. The plugin might be easier, at least in the beginning, since it can record clicks, but Nightwatch's syntax is pretty intuitive. Nightwatch uses a JavaScript object for each test, containing a sequence of commands and assertions.

There is a sample Nightwatch test that works, though I didn't have time to write a DSpace specific one yet.
